feedhenry/fh-messaging
# Prune collections for fhact in fh-messaging

## Jira link's
https://issues.jboss.org/browse/RHMAP-19375

## What
In cases where the fh-messaging pruning job fails on a particular day, we can end up with situations where the collections are not pruned and live on forever or until manual intervention is undertaken.

## Why
Reason : ( FHOPS needs )


## How
The pruning job will now also run for the previous day also, each time it is executed.

## Verification steps

### Info
- The fh-messaging configuration **metrics\_rollup\_job daysToKeep** defaults to `31` days in production
- fh-messaging uses the **fh-reporting** database in Mongo in the MBaaS project.
- The admin password can be retrieved from the **mongodb-admin-password** key inside the **mongodb-keys** config-map in the MBaaS project.

### Verification
- Deploy changes to a cluster.
- Connect to the Mongo primary pod in the replica set in the MBaaS project. eg: `oc rsh mongodb-1-1-wt99t`
- Launch the mongo client: `mongo 127.0.0.1/admin -u admin -p`
- `use fh-reporting` use the database
- `show collections` list the collections, and make note of the **fhact_yyyymmdd** collections
- Connect to one of the fh-messaging pods
- Run the pruning command `bash bin/fh-msg-process $(date '+%F') NO_IMPORT config`
- `show collections` once more on the Mongo replica primary. And make note of the **fhact_yyyymmdd** collections. Collections from dates more than 31 days ago should no longer be listed.
